inc/sitemaps/class-sitemap-image-parser.php

```diff
@@ -62,19 +48,22 @@ public function get_images( $post ) {
 
 		$images = array();
 
-		$content = $post->post_content;
-		$content = '<p><img src="' . $this->image_url( get_post_thumbnail_id( $post->ID ) ) . '" alt="' . $post->post_title . '" /></p>' . $content;
-
-		if ( preg_match_all( '`<img [^>]+>`', $content, $matches ) ) {
-
-			$images = $this->parse_matched_images( $matches[0], $post );
+		if ( ! is_object( $post ) ) {
```

Comment:
Would it be better to check that it's a type of `WP_Post` (if that existed for the minimum version of WP that you support), rather than a general object?